src/studio-header/DesktopHeader.jsx

```diff
@@ -14,9 +14,22 @@ import messages from './Header.messages';
 // Assets
 import { CaretIcon } from './Icons';
 
+// Data
+import StudioApiService from '../data/services/StudioApiService';
+
 class DesktopHeader extends React.Component {
   constructor(props) { // eslint-disable-line no-useless-constructor
     super(props);
+    this.state = {
+      courseNumber: null,
+      org: '',
+      courseTitle: '',
+    };
+  }
+
+  async componentDidMount() {
+    const response = await StudioApiService.getCourseDetailsData(this.props.courseId);
```

Comment:
Not sure how much error handling should be included, and if so, what should be displayed if there is an error.

Reply:
We could fallback to the behavior we have currently - show the course ID with a link back to the course outline?